packages/protocol/contracts/rewards/ProtocolRewards.sol

```diff
@@ -0,0 +1,262 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {EIP712Upgradeable} from "@openzeppelin/contracts-upgradeable/utils/cryptography/EIP712Upgradeable.sol";
+import {IERC20Extended} from "../interfaces/IERC20Extended.sol";
+
+import {SecurityModule} from "../SecurityModule.sol";
+import {IProtocolRewards} from "./IProtocolRewards.sol";
+
+contract ProtocolRewards is
+    IProtocolRewards,
+    SecurityModule,
+    EIP712Upgradeable
+{
+    using SafeERC20 for IERC20Extended;
+
+    bytes32 public constant WITHDRAW_TYPEHASH =
+        keccak256(
+            "Withdraw(address from,address to,uint256 amount,uint256 nonce,uint256 deadline)"
+        );
+
+    bytes32 public constant BLOCK_UNBLOCK_ROLE =
+        keccak256("BLOCK_UNBLOCK_ROLE");
+
+    IERC20Extended public token;
+
+    mapping(address => uint256) public balanceOf;
+    mapping(address => uint256) public nonces;
+    mapping(address => bool) public blockList;
+
+    modifier IfNonBlocked(address _address) {
+        if (_address != address(0) && blockList[_address] == true)
+            revert BLOCKED();
+        _;
+    }
+
+    /**
+     *
+     * @param _token Reward token address.
+     */
+    function initialize(address _token, address _owner) external initializer {
+        __Pausable_init();
+        __EIP712_init("ProtocolRewards", "1");
+
+        if (_token == address(0)) {
+            revert ADDRESS_ZERO();
+        }
+
+        if (_owner == address(0)) {
+            revert ADDRESS_ZERO();
+        }
+
+        token = IERC20Extended(_token);
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _owner);
+    }
+
+    function totalSupply() external view returns (uint256) {
+        return token.balanceOf(address(this));
+    }
+
+    /**
+     * Deposit rewards for  an address.
+     * @param _to Address of reward beneficiary
+     * @param _amount  Total amount to deposit
+     * @param _reason reason of deposit, example value bytes4(keccak256("PROTOCOL_FEE"));     // 0x8e1a55d1
+     * @param _comment Comment
+     */
+    function deposit(
+        address _to,
+        uint256 _amount,
+        bytes4 _reason,
+        string calldata _comment
+    ) external {
```

Comment:
checking if we need to add protection to this deposit function? someone can play around with deposit and withdraw functions

Reply:
Keeping it open as external teams can use this to distribute rewards.